hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -108,6 +108,8 @@ public class CreateTopLevelDagVisitor extends CreateDagVisitorBase<Vertex> {
 
     private final DagBuildContextImpl dagBuildContext;
 
+    private Integer requiredRootPartitionId;
```

Comment:
@Fly-Style @ivanthescientist this would be a good use for processor pruning. currently we have inefficient workaround which requires that all processors are also created on coordinator node, even if no data reside there.
It would be better to run only `RootResultConsumerSink` on coordinator and everything else would be executed where it needs to be run. Jet would take care of sending data to root sink.
WDYT? Either way, it has to be described in TDD (SQL part).

Reply:
II agree with you that it's an ideal scenario, but it requires some time to design and implement processor pruning, which is **very** non-trivial problem.

Reply:
OTOH, refactoring this later may be harder than doing it right from the beginning.

Reply:
We should merge what we have for now. Processor pruning is an entirely different subject.

Reply:
ok, but has to described in TDD

Reply:
TDD is in your branch/PR, we can discuss it there once I transfer the filter analysis section out of docs. Either way its out of scope of this PR since its focuses on the implementation.

Reply:
yes, TDD will be in a separate branch. I just left the comment so we do not forget to describe that aspect.